libraries/AP_HAL_Linux/GPIO_RPI.cpp

```diff
@@ -228,7 +228,9 @@ void GPIO_RPI::write(uint8_t pin, uint8_t value)
 
 void GPIO_RPI::toggle(uint8_t pin)
 {
-    write(pin, !read(pin));
+    uint32_t flag = (1 << pin);
```

Comment:
we need to protect pin going above 32. BCM2835 actually has 54 GPIOs, but it seems we only support  setting/clearing GPSET0/GPCLR0, so just checking for `pin >= GPIO_RPI_MAX_NUMBER_PINS` like is done in the other public functions is good enough.

Also, AFAICS from the datasheet, GPLEV0 should still return a valid value even if the pin is in output mode. Is that not working? In what board?

Reply:
1- I added a check for pin > 32

2- It is not working on raspberry pi zero !